loader/loader_test.go

```diff
@@ -0,0 +1,423 @@
+			Expect(returnedObject.Name).To(Equal(environment.Name))
+		})
+	})
+
+	Context("When calling GetRelease", func() {
+		It("returns the requested release", func() {
+			returnedObject, err := loader.GetRelease(ctx, k8sClient, release.Name, release.Namespace)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(returnedObject).NotTo(Equal(&v1alpha1.Release{}))
+			Expect(returnedObject.Name).To(Equal(release.Name))
+		})
+	})
+
+	Context("When calling GetReleasePipelineRun", func() {
+		It("returns a PipelineRun if the labels match with the release data", func() {
+			returnedObject, err := loader.GetReleasePipelineRun(ctx, k8sClient, release)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(returnedObject).NotTo(Equal(&v1beta1.PipelineRun{}))
+			Expect(returnedObject.Name).To(Equal(pipelineRun.Name))
+		})
+
+		It("fails to return a PipelineRun if the labels don't match with the release data", func() {
+			modifiedRelease := release.DeepCopy()
+			modifiedRelease.Name = "non-existing-release"
+
+			returnedObject, err := loader.GetReleasePipelineRun(ctx, k8sClient, modifiedRelease)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(returnedObject).To(BeNil())
+		})
+	})
+
+	Context("When calling GetReleasePlan", func() {
+		It("returns the requested release plan", func() {
+			returnedObject, err := loader.GetReleasePlan(ctx, k8sClient, release)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(returnedObject).NotTo(Equal(&v1alpha1.ReleasePlan{}))
+			Expect(returnedObject.Name).To(Equal(releasePlan.Name))
+		})
+	})
+
+	Context("When calling GetReleaseStrategy", func() {
+		It("returns the requested release strategy", func() {
+			returnedObject, err := loader.GetReleaseStrategy(ctx, k8sClient, releasePlanAdmission)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(returnedObject).NotTo(Equal(&v1alpha1.ReleaseStrategy{}))
+			Expect(returnedObject.Name).To(Equal(releaseStrategy.Name))
+		})
+	})
+
+	Context("When calling GetSnapshot", func() {
+		It("returns the requested snapshot", func() {
+			returnedObject, err := loader.GetSnapshot(ctx, k8sClient, release)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(returnedObject).NotTo(Equal(&applicationapiv1alpha1.Snapshot{}))
+			Expect(returnedObject.Name).To(Equal(snapshot.Name))
+		})
+	})
+
+	Context("When calling GetSnapshotEnvironmentBinding", func() {
+		It("returns a snapshot environment binding if the environment field value matches the release plan admission one", func() {
+			returnedObject, err := loader.GetSnapshotEnvironmentBinding(ctx, k8sClient, releasePlanAdmission)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(returnedObject).NotTo(Equal(&applicationapiv1alpha1.SnapshotEnvironmentBinding{}))
+			Expect(returnedObject.Name).To(Equal(snapshotEnvironmentBinding.Name))
+		})
+
+		It("fails to return a snapshot environment binding if the environment field value doesn't match the release plan admission one", func() {
+			modifiedReleasePlanAdmission := releasePlanAdmission.DeepCopy()
```

Comment:
this should be deleted like line 95 right?

Reply:
Yes! Just kidding. Like before, no object is created here, so no need to delete anything.

Reply:
This is what I get for being thorough :roll_eyes: